src/metatrain/gap/model.py

```diff
@@ -393,141 +339,47 @@ class _SorKernelSolver:
     def __init__(
         self,
         KMM: np.ndarray,
-        regularizer: float = 1.0,
-        jitter: float = 0.0,
-        solver: str = "RKHS",
-        relative_jitter: bool = True,
     ):
-        self.solver = solver
         self.KMM = KMM
-        self.relative_jitter = relative_jitter
 
         self._nM = len(KMM)
-        if self.solver == "RKHS" or self.solver == "RKHS-QR":
-            self._vk, self._Uk = scipy.linalg.eigh(KMM)
-            self._vk = self._vk[::-1]
-            self._Uk = self._Uk[:, ::-1]
-        elif self.solver == "QR" or self.solver == "solve" or self.solver == "lstsq":
-            # gets maximum eigenvalue of KMM to scale the numerical jitter
-            self._KMM_maxeva = scipy.sparse.linalg.eigsh(
-                KMM, k=1, return_eigenvectors=False
-            )[0]
-        else:
-            raise ValueError(
-                f"Solver {solver} not supported. Possible values "
-                "are 'RKHS', 'RKHS-QR', 'QR', 'solve' or lstsq."
-            )
-        if relative_jitter:
-            if self.solver == "RKHS" or self.solver == "RKHS-QR":
-                self._jitter_scale = self._vk[0]
-            elif (
-                self.solver == "QR" or self.solver == "solve" or self.solver == "lstsq"
-            ):
-                self._jitter_scale = self._KMM_maxeva
+        self._vk, self._Uk = scipy.linalg.eigh(KMM)
+        self._vk = self._vk[::-1]
+        self._Uk = self._Uk[:, ::-1]
+
+        self._nM = len(np.where(self._vk > 0)[0])
+        self._PKPhi = self._Uk[:, : self._nM] * 1 / np.sqrt(self._vk[: self._nM])
+
+    def fit(
+        self, KNM: Union[torch.Tensor, np.ndarray], Y: Union[torch.Tensor, np.ndarray]
```

Comment:
Why do we allow numpy objects here?

Reply:
the training is done in numpy, with numpy arrays on cpu

Reply:
Ahh yes - sure kernels. Works!